catkin_tools/verbs/catkin_clean/cli.py

```diff
@@ -14,178 +14,367 @@
-
-                        if os.path.exists(ccache_path):
-                            print(" - Removing %s" % ccache_path)
-                            os.remove(ccache_path)
-                            needs_force = True
+            if not opts.dry_run:
+                shutil.rmtree(ctx.build_space_abs)
+
+        # Setup file removal
+        if opts.setup_files:
+            if devel_exists:
+                print("[clean] Removing setup files from develspace: %s" % ctx.devel_space_abs)
+                opts.packages.append('catkin')
+                opts.packages.append('catkin_tools_prebuild')
             else:
-                print("[clean] No buildspace exists, no CMake caches to clear.")
+                print("[clean] No develspace exists, no setup files to clean.")
+
+        # Clean log files
+        if opts.logs and logs_exists:
+            print("[clean] Removing log space: {}".format(ctx.log_space_abs))
+            if not opts.dry_run:
+                shutil.rmtree(ctx.log_space_abs)
+
+        # Find orphaned packages
+        if ctx.link_devel and not any([opts.build, opts.devel]):
+            if opts.orphans:
+                if os.path.exists(ctx.build_space_abs):
+                    print("[clean] Determining orphaned packages...")
+
+                    # Get all existing packages in source space and the
+                    # Suppress warnings since this is looking for packages which no longer exist
+                    found_source_packages = [
+                        pkg.name for (path, pkg) in
+                        find_packages(ctx.source_space_abs, warnings=[]).items()]
+                    built_packages = [
+                        pkg.name for (path, pkg) in
+                        find_packages(ctx.package_metadata_path(), warnings=[]).items()]
+
+                    # Look for orphaned products in the build space
+                    orphans = [p for p in built_packages
+                               if (p not in found_source_packages and p !=
+                                   'catkin_tools_prebuild')]
+
+                    if len(orphans) > 0:
+                        opts.packages.extend(list(orphans))
+                    else:
+                        print("[clean] No orphans in the workspace.")
+                else:
+                    print("[clean] No buildspace exists, no potential for orphans.")
+
+            # Remove specific packages
+            if len(opts.packages) > 0:
+
+                try:
+                    # Clean the packages
+                    needs_force = clean_packages(
+                        ctx,
+                        opts.packages,
+                        opts.dependants,
+                        opts.verbose,
+                        opts.dry_run)
+                except KeyboardInterrupt:
+                    wide_log("[build] User interrupted!")
+                    event_queue.put(None)
```

Comment:
`event_queue` is not defined here, perhaps you forgot to pass it when moving this code into a function?